orchagent/intfsorch.cpp

```diff
@@ -623,10 +624,20 @@ void IntfsOrch::removeIp2MeRoute(sai_object_id_t vrf_id, const IpPrefix &ip_pref
     }
 }
 
-void IntfsOrch::addDirectedBroadcast(const Port &port, const IpAddress &ip_addr)
+void IntfsOrch::addDirectedBroadcast(const Port &port, const IpPrefix &ip_prefix)
 {
     sai_status_t status;
     sai_neighbor_entry_t neighbor_entry;
+    IpAddress ip_addr;
+
+    /* For /31 and /32 v4 subnets, there is no broadcast address, hence don't
+     * add a broadcast route. */
+    if ((ip_prefix.isV4()) && (ip_prefix.getMaskLength() > 30))
```

Comment:
Another option is to modify this check as` if (!(ip_prefix.isV4()) || (ip_prefix.getMaskLength() > 30))` and edit the comment!.

Reply:
Done.